app/src/main/java/app/revanced/integrations/youtube/patches/components/DescriptionComponentsFilter.java

```diff
@@ -45,21 +46,44 @@ public DescriptionComponentsFilter() {
                 "transcript_section"
         );
 
+        macroMarkersCarousel = new StringFilterGroup(
+                null,
+                "macro_markers_carousel.eml"
+        );
+
+        macroMarkersCarouselGroupList.addAll(
+                new ByteArrayFilterGroup(
+                        Settings.HIDE_CHAPTERS_SECTION,
+                        "chapters_horizontal_shelf"
+                ),
+                new ByteArrayFilterGroup(
+                        Settings.HIDE_KEY_CONCEPTS_SECTION,
+                        "learning_concept_macro_markers_carousel_shelf"
+                )
+        );
+
         addPathCallbacks(
                 attributesSection,
-                chaptersSection,
                 infoCardsSection,
                 podcastSection,
-                transcriptSection
+                transcriptSection,
+                macroMarkersCarousel
         );
     }
 
-
     @Override
     boolean isFiltered(@Nullable String identifier, String path, byte[] protobufBufferArray,
                        StringFilterGroup matchedGroup, FilterContentType contentType, int contentIndex) {
         if (exceptions.matches(path)) return false;
 
+        if (matchedGroup == macroMarkersCarousel) {
+            if (contentIndex == 0)
+                return false;
+        }
+                  if (!macroMarkersCarouselGroupList.check(protobufBufferArray).isFiltered()) {
```

Comment:
Why is this check outside of the `macroMarkersCarousel` check?

Reply:
Is it fine now?

Reply:
Done.